### PR TITLE
Ability to configure client to send default headers #284 at 2.2.x

### PR DESCRIPTION
**Pull request by IlyasYOY**
Hello!

I've tried to implement feature I requested in Issue #284
TBH I just updated my old solution.

I've got some likes there so I decided to post this PR.

Thanks you for your work!

> Reopened from 2.2.x instead of master. Source  PR: https://github.com/spring-cloud/spring-cloud-openfeign/pull/415

### CODE REVIEW

**Comment by OlgaMaciaszek** [comment]
Fixes gh-284.

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/FeignClientUsingPropertiesTests.java [hunk 1]
```diff
@@ -253,18 +321,32 @@ public String foo(HttpServletRequest request) throws IllegalAccessException {
...
 		}
 
-		@RequestMapping(value = "/form", method = RequestMethod.POST,
+		@PostMapping(path = "/form",
 				consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
 		public String form(HttpServletRequest request) {
 			return request.getParameter("form");
 		}
 
+		@GetMapping(path = "/eggs")
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/FeignClientUsingPropertiesTests.java:336]: @IlyasYOY Please also adujst the test names, endpoints, etc. as requested [here](https://github.com/spring-cloud/spring-cloud-openfeign/pull/415#discussion_r514210725).

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Thanks, @IlyasYOY This is nearly ready. Have added one more comment.

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/FeignClientUsingPropertiesTests.java (hunk 1 above)
Comment (IlyasYOY) [reply spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/FeignClientUsingPropertiesTests.java]: Fixed + also replaced `Thread.sleep(2000L)` with more verbose `TimeUnit.SECONDS.sleep(2)`

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/FeignClientUsingPropertiesTests.java [hunk 2]
```diff
@@ -159,6 +185,35 @@ public void testForm() {
 		assertThat(response).isEqualTo("Data");
 	}
 
+	@Test
+	public void testEggs() {
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/FeignClientUsingPropertiesTests.java:189]: It's still here. Can you just do a text-search on `eggs` (it does not have any meaning in the domain)? Once fixed, it'll be ready to go.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Thanks, @IlyasYOY. Nearly done. Please see the new comment.

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/FeignClientUsingPropertiesTests.java (hunk 2 above)
Comment (IlyasYOY) [reply spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/FeignClientUsingPropertiesTests.java]: I'm so sorry, I just missed them (used case sensitive search)

Fixed them, now it looks good. Variables and methods have concise and sensible names.

**Review by OlgaMaciaszek — APPROVED** [review]

**Pull request merged**